azurerm/resource_arm_storage_account.go

```diff
@@ -663,10 +699,45 @@ func resourceArmStorageAccountRead(d *schema.ResourceData, meta interface{}) err
 		}
 
 		if endpoints := props.PrimaryEndpoints; endpoints != nil {
-			d.Set("primary_blob_endpoint", endpoints.Blob)
-			d.Set("primary_queue_endpoint", endpoints.Queue)
-			d.Set("primary_table_endpoint", endpoints.Table)
-			d.Set("primary_file_endpoint", endpoints.File)
+			if v := endpoints.Blob; v != nil {
+				d.Set("primary_blob_endpoint", endpoints.Blob)
+				if u, err := url.Parse(*v); err != nil {
```

Comment:
We should either be checking for an error and bubbling it up or refactor this to not use `url.Parse()`